File: week9_LabBG75-BG75-OLQ_9-BDG/b.c
#include <math.h>
#include <stdio.h>
#define ll long long

ll binarySearch(ll m) {
  ll low = 1, high = m, mid;
  while (low < high) {
    mid = low + (high - low) / 2;
    ll bruh = (mid * (mid + 1) * (2 * mid + 1)) / 6;
    if (bruh == m)
      return mid;
    else if (bruh < m)
      low = mid + 1;
    else
      high = mid;
  }
  return low;
}

void solve() {
  ll m;
  scanf("%lld", &m);
  printf("%lld\n", binarySearch(m));
}

int main() {
  int t;
  scanf("%d", &t);
  getchar();
  for (int i = 1; i <= t; i++) {
    printf("Case #%d: ", i);
    solve();
  }
}

File: week9_LabBG75-BG75-OLQ_9-BDG/b.py
def binary_search(m: int) -> int:
    low, high = 1, m
    while low < high:
        mid = (low + high) // 2
        bruh = (mid * (mid + 1) * (2 * mid + 1)) // 6
        if bruh == m:
            return mid
        elif bruh < m:
            low = mid + 1
        else:
            high = mid
    return low

def solve(m: int):
    print(binary_search(m))

t = int(input())
for i in range(1, t+1):
    print("Case #{}: ".format(i), end="")
    m = int(input())
    solve(m)
